DBOS-project/apiary
# Postgres Support

This PR adds Postgres support through JDBC.

To start a Postgres instance through Docker:
```
bash scripts/initialize_postgres_docker.sh
```

Then run Postgres tests:
```
mvn -Dtest=PostgresTests package
```

It should succeed. Next steps: 
1) Test transactional function composition works. Do we support tested transactions in Postgres?
2) Clean up implementation of Cockroach and SQLite. Their implementations are not compatible with function compositions.
3) Add Postgres to the worker executable and add benchmarks.
4) Add a better client library to register functions and tables.
